host_source.go

```diff
@@ -96,6 +96,11 @@ func (r *ringDescriber) GetHosts() (hosts []HostInfo, partitioner string, err er
 		localHost.Peer = addr
 	}
 
+	// Make sure we didn't screw up discovering our local peer.
+	if localHost.Peer == "" || localHost.Peer == "0.0.0.0" {
+		return nil, "", ErrHostQueryFailed
```

Comment:
If this is the case then we should use conn.Addr() which is less bad than just never discovering